casper/src/main/scala/coop/rchain/casper/engine/BlockRetriever.scala

```diff
@@ -230,7 +230,7 @@ object BlockRetriever {
                         s"Remain waiting: ${waitingListTail.map(_.endpoint.host).mkString(", ")}."
                     )
                 _  <- CommUtil[F].requestForBlock(nextPeer, hash)
-                ts <- Time[F].currentMillis
+                ts <- Timer[F].clock.realTime(TimeUnit.MILLISECONDS)
```

Comment:
I'm curious why do you choose Timer and not Clock? I see that Timer is removed in cats-effects 3, but Clock is still there.

Reply:
I was only thinking about the first step to remove custom Time first.
- #3730

But your point makes sense. We could use only Clock in these situations and even create extensions to mimic v3 for easier migration.

Timer is renamed to Temporal in v3 and also extends Concurrent.